binding-macro/src/service.rs

```diff
@@ -87,81 +87,99 @@ pub fn gen_service_code(_: TokenStream, item: TokenStream) -> TokenStream {
 
     TokenStream::from(quote! {
         impl #impl_generics protocol::traits::Service for #service_ident #ty_generics #where_clause {
-            fn genesis_(&mut self, _payload: String) -> protocol::ProtocolResult<()> {
+            fn genesis_(&mut self, _payload: String) {
                 #genesis_body
             }
 
-            fn hook_before_(&mut self, _params: &ExecutorParams) -> protocol::ProtocolResult<()> {
+            fn hook_before_(&mut self, _params: &ExecutorParams) {
                 #hook_before_body
             }
 
-            fn hook_after_(&mut self, _params: &ExecutorParams) -> protocol::ProtocolResult<()> {
+            fn hook_after_(&mut self, _params: &ExecutorParams) {
                 #hook_after_body
             }
 
-            fn tx_hook_before_(&mut self, _ctx: ServiceContext) -> ProtocolResult<()> {
+            fn tx_hook_before_(&mut self, _ctx: ServiceContext) {
                 #tx_hook_before_body
             }
 
-            fn tx_hook_after_(&mut self, _ctx: ServiceContext) -> ProtocolResult<()> {
+            fn tx_hook_after_(&mut self, _ctx: ServiceContext) {
                 #tx_hook_after_body
             }
 
-            fn read_(&self, ctx: protocol::types::ServiceContext) -> protocol::ProtocolResult<String> {
+            fn read_(&self, ctx: protocol::types::ServiceContext) -> ServiceResponse<String> {
                 let service = ctx.get_service_name();
                 let method = ctx.get_service_method();
 
                 match method {
                     #(#list_read_name => {
-                        let payload: #list_read_payload = serde_json::from_str(ctx.get_payload())
-                                .map_err(|e| protocol::traits::BindingMacroError::JsonParse(e))?;
-                        let res = self.#list_read_ident(ctx, payload)?;
-                        let res_str = serde_json::to_string(&res).map_err(|e| protocol::traits::BindingMacroError::JsonParse(e))?;
-                        if res_str.as_str() == "null" {
-                            Ok("".to_owned())
+                        let payload_res: Result<#list_read_payload, _> = serde_json::from_str(ctx.get_payload());
+                        if payload_res.is_err() {
+                            return ServiceResponse::<String>::from_error(1, "service macro decode payload failed".to_owned());
```

Comment:
Don't call it service macro because this is a runtime error and not macro error.

Reply:
fixed:
```rust
return ServiceResponse::<String>::from_error(1, "decode service payload failed".to_owned());
```